### DISCUSSION TITLE
Error in initializing org.owasp.esapi.logging.java.JavaLogFactory.

### DISCUSSION THREAD

**Discussion by Somdutta** [ESAPI/esapi-java-legacy#750] [Q&A]
Hi,

I am using esapi-2.5.0.0.jar for my project. I have set ESAPI.Logger=org.owasp.esapi.logging.java.JavaLogFactory in my ESAPI.properties file. However, I am getting the error "java.lang.NoClassDefFoundError: Could not initialize class org.owasp.esapi.logging.java.JavaLogFactory".

Could you please advice.

Thanks,
Somdutta.

**Comment by jeremiahjstacey** [comment]
1.  If creating a bug, please use the bug template provided to help us help you better.
2. This is not a bug.  This is a problem you're having integrating ESAPI.  These types of questions are better posed in [StackOverflow](https://stackoverflow.com/), asked as a project [Discussion](https://github.com/ESAPI/esapi-java-legacy/discussions),  or communicated through one of the mailing lists identified in the project [README](https://github.com/ESAPI/esapi-java-legacy#references-where-to-find-more-information-on-esapi).
    - The names of the 2 Google Groups are "[esapi-project-users](mailto:esapi-project-users@owasp.org)" and "[esapi-project-dev](mailto:esapi-project-dev@owasp.org)", which you may POST to after you subscribe to them via "[Subscribe to ESAPI Users list](https://groups.google.com/a/owasp.org/forum/#!forum/esapi-project-users/join)" and "[Subscribe to ESAPI Developers list](https://groups.google.com/a/owasp.org/forum/#!forum/esapi-project-dev/join)" respectively
3. https://github.com/ESAPI/esapi-java-legacy/wiki/Configuration-Reference:-JavaLogFactory#debugging-failed-startup  You problem is most likely that you are missing esapi-java-logging.properties.
    - PLEASE NOTE the [VERSION COMPATIBILTY NOTICE](https://github.com/ESAPI/esapi-java-legacy/wiki/Configuration-Reference:-JavaLogFactory#version-compatibility-notice) on that document
